pkgs/misc/emulators/melonDS/default.nix

```diff
@@ -0,0 +1,29 @@
+{ stdenv, fetchFromGitHub, cmake, pkgconfig, SDL2, gtk3, curl, libpcap }:
+
+stdenv.mkDerivation {
+  pname = "melonDS";
+  version = "0.8.3";
+
+  src = fetchFromGitHub {
+    owner = "Arisotura";
+    repo = "melonDS";
+    rev = "e8f4735c7fb4d9fc142a4571e99672ca750be66b";
+    sha256 = "1lqmfwjpkdqfkns1aaxlp4yrg6i0r66mxfr4rrj7b5286k44hqwn";
+  };
+
+  enableParallelBuilding = true;
+  nativeBuildInputs = [ cmake pkgconfig ];
+  buildInputs = [ SDL2 gtk3 curl libpcap ];
+
+  postInstall = ''
+    install -Dm644 romlist.bin "$out/share/melonDS/romlist.bin"
```

Comment:
This file is needed for saving to work right, but there's currently no way to set its path without adding a patch. I include it here so that users can symlink it into their directory along with the other binaries required.

Reply:
https://nixos.org/nixpkgs/manual/#sec-patches

You can include a nixpkgs-specific patch in this directory too if you think that would be useful/better for users. If it's not nixpkgs specific then it's preferred you submit the patch to upstream if possible. Because keep in my as maintainer you would be expected to maintain any nixpkgs-specific patches you include.

Reply:
I'm going to work on this upstream when I have time. Currently upstream makes a lot of assumptions which would make a patch a bit unclean.